src/lib/routes.js

```diff
@@ -26,23 +27,39 @@ module.exports = server => {
             next(new server.errors.InternalServerError(err));
           });
       }
-
-      next(new server.errors.BadRequestError(`Make sure the date format is '${dateFormat}' (time is optional)`));
+      next(new server.errors.BadRequestError(`Make sure the date format is "${dateFormat}" (time is optional)`));
     } else {
-      next(new server.errors.BadRequestError('Must provide a \'from\' date parameter, and optionally a \'to\' date'));
+      next(new server.errors.BadRequestError('Must provide a "from" date parameter, and optionally a "to" date'));
     }
   });
 
+  server.get('/dashboard', (req, res) => {
+    dashboardModel()
+      .then(props => res.render(LevDashboard, props))
+      .catch(err => server.log.error(err));
+  });
+
+  server.get('/dashboard/data', (req, res, next) => { // eslint-disable-line consistent-return
+    return dashboardModel()
+      .then(data => JSON.stringify(data, null, 2))
+      .then(res.end.bind(res))
+      .catch(err => {
+        server.log.error(err);
+        next(new server.errors.InternalServerError(err));
+      });
+  });
+
   server.get('/*', (req, res) => {
-    let fromDate = req.query && req.query.from && moment(req.query.from, dateFormat) || moment().startOf('month');
-    fromDate = (fromDate.isValid() ? fromDate : moment()).format(dateFormat);
+    const fromDate = req.query && req.query.from && moment(req.query.from, dateFormat);
     const toDate = req.query && req.query.to && moment(req.query.to, dateFormat);
 
-    model(fromDate, toDate)
-      .then(props => res.render(report, props))
+    model(
+      fromDate.isValid() ? fromDate.format(dateFormat) : moment().startOf('month'),
+      toDate.isValid() && toDate.format(dateFormat))
+      .then(props => res.render(LevReport, props))
       .catch(err => {
         server.log.error(err);
-        res.render(report);
+        res.render(LevReport);
```

Comment:
Same as above.

Probably should be:
```js
next(err);
```

Reply:
Validation errors should be handled separately really. (Another ticket?)

Reply:
fixed

Reply:
fixed